src/rem/element.tsx

```diff
@@ -167,6 +167,16 @@ const RemCalculator: FC<{
           type: error.type,
         }),
       );
+
+    shadowRoot.dispatchEvent(
+      new CustomEvent('bi-calculator-submitted', {
+        bubbles: true,
+        composed: true,
+        detail: {
+          formData: formValues,
```

Comment:
The chosen upgrade is not in `formValues`; it's in the separate variable `upgradeValue`. (This is a little historical artifact.) So you probably want to do:

`formData: {...formValues, upgrade: upgradeValue }`